integration/entity.go

```diff
@@ -105,12 +107,13 @@ func (e *Entity) SameAs(b *Entity) bool {
 	if e.Metadata == nil || b.Metadata == nil {
 		return false
 	}
-
 	return e.Metadata.EqualsTo(b.Metadata)
 }
 
 // NewMetricSet returns a new instance of Set with its sample attached to the integration.
 func (e *Entity) NewMetricSet(eventType string, nameSpacingAttributes ...attribute.Attribute) *metric.Set {
+	e.lock.Lock()
+	defer e.lock.Unlock()
```

Comment:
I think I'm lacking some context to understand the issue. Here we are moving the lock lines so we protect the modification of the metadata. when a new MetricSet is added. Then, we also lock the mutex when getting the entity Key. What problem does this solve?

Reply:
It was not creating deadlock because it's two different mutexes, I basically created a new mutex for the entityMetadata and the one move upward is from the entity.
But anyway, moving this one upward was a inherited from my first solution where I was only using one mutex and now I removed the change moving it upward, and with the mutex in the key() method is enough to solve the race condition.